data/actions/scripts/other/fluids.lua

```diff
@@ -26,58 +26,58 @@ local fluidMessage = {
 local distillery = {[5513] = 5469, [5514] = 5470}
 
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
-	local targetItemType = ItemType(target.itemid)
+	local targetItemType = ItemType(target:getId())
 	if targetItemType and targetItemType:isFluidContainer() then
-		if target.type == 0 and item.type ~= 0 then
-			target:transform(target:getId(), item.type)
+		if target.type == 0 and item:getCount() ~= 0 then
+			target:transform(target:getId(), item:getCount())
 			item:transform(item:getId(), 0)
 			return true
-		elseif target.type ~= 0 and item.type == 0 then
+		elseif target.type  ~= 0 and item:getCount() == 0 then
 			target:transform(target:getId(), 0)
 			item:transform(item:getId(), target.type)
 			return true
 		end
 	end
 
-	if target.itemid == 1 then
-		if item.type == 0 then
+	if target:getId() == 1 then
+		if item:getCount() == 0 then
 			player:sendTextMessage(MESSAGE_STATUS_SMALL, "It is empty.")
-		elseif target.uid == player.uid then
-			if table.contains({3, 15, 43}, item.type) then
+		elseif target:getId() == player:getId() then
+			if table.contains({3, 15, 43}, item:getCount()) then
 				player:addCondition(drunk)
-			elseif item.type == 4 then
+			elseif item:getCount() == 4 then
 				player:addCondition(poison)
-			elseif item.type == 7 then
+			elseif item:getCount() == 7 then
 				player:addMana(math.random(50, 150))
 				fromPosition:sendMagicEffect(CONST_ME_MAGIC_BLUE)
-			elseif item.type == 10 then
+			elseif item:getCount() == 10 then
 				player:addHealth(60)
 				fromPosition:sendMagicEffect(CONST_ME_MAGIC_BLUE)
 			end
-			player:say(fluidMessage[item.type] or "Gulp.", TALKTYPE_MONSTER_SAY)
+			player:say(fluidMessage[item:getCount()] or "Gulp.", TALKTYPE_MONSTER_SAY)
 			item:transform(item:getId(), 0)
 		else
-			Game.createItem(2016, item.type, toPosition):decay()
+			Game.createItem(2016, item:getCount(), toPosition):decay()
 			item:transform(item:getId(), 0)
 		end
 	else
 		local fluidSource = targetItemType and targetItemType:getFluidSource() or 0
 		if fluidSource ~= 0 then
 			item:transform(item:getId(), fluidSource)
-		elseif table.contains(distillery, target.itemid) then
-			local tmp = distillery[target.itemid]
+		elseif table.contains(distillery, target:getId()) then
+			local tmp = distillery[target:getId()]
 			if tmp then
 				item:transform(item:getId(), 0)
 			else
 				player:sendCancelMessage("You have to process the bunch into the distillery to get rum.")
 			end
-		elseif item.type == 0 then
+		elseif item:getCount() == 0 then
 			player:sendTextMessage(MESSAGE_STATUS_SMALL, "It is empty.")
 		else
 			if toPosition.x == CONTAINER_POSITION then
 				toPosition = player:getPosition()
 			end
-			Game.createItem(2016, item.type, toPosition):decay()
+			Game.createItem(2016, item:getCount(), toPosition):decay()
```

Comment:
here I think it is creating with subtype as well

Reply:
Correct. Item id 2016 is pool, I must use type and not quantity.